types/types.d.ts

```diff
@@ -26,8 +26,11 @@ interface SessionData extends ExpressSessionData {
   /** Updates the `expires` property of the session. */
   touch(): void;
 
-  /** Regenerates the session by generating a new `sessionId`. */
-  regenerate(): void;
+  /**
+   * Regenerates the session by generating a new `sessionId`.
+   * @param request Optional, but REQUIRED if idGenerator uses it
+   */
+  regenerate(request?: Fastify.FastifyRequest): void;
```

Comment:
```suggestion
  /**
   * Regenerates the session by generating a new `sessionId`.
   * @param request optional, but __REQUIRED__ if the `idGenerator` function used to generate new session IDs uses it.
   */
  regenerate(request?: Fastify.FastifyRequest): void;
```

Reply:
One other way to type the `regenerate` method we can explore would be something like this:
```ts
// With this we can get the parameters of a function type in a tuple
type Parameters<T> = T extends (...args: infer P) => any ? P : never;

// and by typing regenerate like this it will accept the same arguments (and types) that `idGenerator` accepts
// NB: if we add another argument to `idGenerator` it will be added to `regenerate` too.
regenerate(...args: Parameters<FastifySessionPlugin.Options['idGenerator']>): void;
```

cc @fastify/typescript WDYT ?